plugin/storage/es/spanstore/reader.go

```diff
@@ -128,23 +122,30 @@ func NewSpanReader(p SpanReaderParams) *SpanReader {
 	if p.UseReadWriteAliases {
 		maxSpanAge = rolloverMaxSpanAge
 	}
+
+	makeName := func(prefix, index string) string {
+		if prefix != "" {
+			return prefix + indexPrefixSeparator + index
```

Comment:
looking at mapping.go I think this should be a bit smarter, e.g.
```suggestion
		if prefix != "" {
			if !prefix.endsWith(indexPrefixSeparator) { prefix += indexPrefixSeparator }
			return prefix + index
```